common/fs/src/cache/mod.rs

```diff
@@ -386,15 +386,15 @@ where
         &self,
         entry: &Entry<T>,
         paths: &mut Vec<PathBuf>,
-        mut components: Vec<String>,
+        mut components: Vec<OsString>,
     ) {
-        let mut base_components: Vec<String> = into_components(&self.resolve_direct_path(entry))
+        let mut base_components: Vec<OsString> = into_components(&self.resolve_direct_path(entry))
             .iter()
-            .map(|x| x.to_str().unwrap().into())
+            .map(|x| x.as_os_str().into())
             .collect(); // build all the components of the path leading up to the current entry
```

Comment:
`into_components()` returns a `Vec<OsString>` so I think you can just omit the entire `.iter().map().collect()`

Reply:
pushed up the change myself

Reply:
Good spot. I'm good with this if you are